File: katalyze/src/main/java/messageHandlers/LanguageHandler.java
package messageHandlers;

import legacyfeed.SimpleMessage;
import model.Language;
import org.apache.log4j.Logger;

public class LanguageHandler extends SingleMessageHandler {

	static Logger logger = Logger.getLogger(LanguageHandler.class);
	
	public LanguageHandler() { super("language"); }
		
	public void process(SimpleMessage message) {
		int id = message.getInt("id");
		String languageName = message.get("name").trim();
	    Language language = new Language(Integer.toString(id), languageName);
		contest.addLanguage(language);
	}

}

File: katalyze/src/main/kotlin/messageHandlers/LanguageHandler.kt
package messageHandlers

import legacyfeed.SimpleMessage
import model.Language
import org.apache.logging.log4j.kotlin.logger

class LanguageHandler : SingleMessageHandler("language") {
    override fun process(message: SimpleMessage) {
        val id = message.getInt("id")
        val languageName = message["name"].trim { it <= ' ' }
        val language = Language(id.toString(), languageName)
        contest!!.addLanguage(language)
    }

    companion object {
        private val logger = logger()
    }
}